googlemock/include/gmock/gmock-actions.h

```diff
@@ -842,7 +841,7 @@ class IgnoreResultAction {
     typedef typename internal::Function<F>::Result Result;
 
     // Asserts at compile time that F returns void.
-    CompileAssertTypesEqual<void, Result>();
+    static_assert(std::is_same<void, Result>::value, "Result type should be void.");
```

Comment:
It's better if you changed it to:
```suggestion
    static_assert(std::is_void<Result>::value, "Result type should be void.");
```